oxide-auth-rocket/src/lib.rs

```diff
@@ -58,9 +58,11 @@ impl<'r> OAuthRequest<'r> {
     /// Create the request data from request headers.
     ///
     /// Some oauth methods need additionally the body data which you can attach later.
-    pub fn new<'a>(request: &'a Request<'r>) -> Self {
-        let query = request.uri().query().unwrap_or("");
-        let query = match serde_urlencoded::from_str(query) {
+    pub fn new(request: &Request<'_>) -> Self {
+        let default_query_uri = uri!("?b");
+        let default_query = default_query_uri.query().unwrap();
+        let query = request.uri().query().unwrap_or(default_query);
```

Comment:
This will need some justification/explanation as in additional comments. Why this specific query string instead of the empty one previously? Is there some type change that requires this—in a first look it seems that maybe a simple clone `&query.to_string()` would have sufficed? Justifiy the unwraps as well, please.

Reply:
Will try more tomorrow, there have been changes to uri/query that broke the previous code.